Dev mod api, 16 BIT I/O, PORTS, ASCII text display(32768 charachters or 64 charachters, 16 or 8Bit)

parshwa282011 opened the pull request:
My goal is to create a Mod API in a way that requires less work and has access to most of the functions in main code, while also adding ports which I want to do with http or WebSockets.

For Ascii text display have it be customizable for some characters not all.
also for key detector add more keys like delete, enter, and etc. Also make it possible for key combinations to also be made like ctrl + c might trigger something

Idea for Mod Api is to have some functions be sent during initialization so that the are registered with the other chips so basically when imported mod is ran it send the init function and the code runs that to init the chip, and the loop function is ran in a during rendering if that chip is running then the loop function is called

- [x] 16 Bit I/O
- [x] 16 Bit Mergers/Splitters
- [x] Displays use 8 Bit for color
- [x] Http server for input and output ports chips for 1-16bit ports
- [x] WebSocket server for input and output ports chips for 1-16bit ports
- [x] Ascii text display 8Bit
- [x] Ascii text display 16Bit
- [x] Key press detection made better
- [ ] Mod Api

parshwa282011 committed:
trying to make a 16 bit inputs and outputs

parshwa282011 committed:
trying to make a 16 bit inputs and outputs

parshwa282011 committed:
Merge branch 'SebLague:main' into main

parshwa282011 committed:
Merge branch 'SebLague:main' into main

Commit:
RGBLED working + added 16 Bit Bus, and 16 Bit IN?OUT the 16 Bit merges and Splits arent working yet

Commit:
RGBLED working + added 16 Bit Bus, and 16 Bit IN?OUT the 16 Bit merges and Splits arent working yet

Commit:
ALL 16-BIT merge and Splits work now

Commit:
ALL 16-BIT merge and Splits work now

Commit:
fixed a few bugs

Commit:
fixed a few bugs

parshwa282011 committed:
Merge branch 'main' into merge

parshwa282011 committed:
Merging for new updates

Merge

parshwa282011 committed:
changed 0b11111111111111111111111111111111 to 0xFFFFFFFF as suggested

parshwa282011 committed:
changed 0b11111111111111111111111111111111 to 0xFFFFFFFF as suggested

parshwa282011 committed:
Removing vscode files

parshwa282011 committed:
Removing vscode files

parshwa282011 committed:
Create main.yml

parshwa282011 committed:
Create main.yml

parshwa282011 committed:
My mistake i ment for this to be on test branch

parshwa282011 committed:
My mistake i ment for this to be on test branch

sk337 commented on ProjectSettings/ProjectVersion.txt:
i cant talk for SebLauge. but i would leave it to him to update the version of unity used

parshwa282011 replied:
oh right its just the version i was using i think i will reverse this when its at the end

pk3zip reviewed (changes requested):
I believe that you that you should do one thing at a time as a modding API in already on the way (#459), and to make the reviewing process easier it common for pull requests to only add 1 large feature at a time. For example you would make 1 pull request for 16Bit IO/Ports, 1 for a ASCII test Display Etc. I would recommend to you to change this Pull request to be just for the 16Bit io.

parshwa282011 commented:
> I believe that you that you should do one thing at a time as a modding API in already on the way (#459), and to make the reviewing process easier it common for pull requests to only add 1 large feature at a time. For example you would make 1 pull request for 16Bit IO/Ports, 1 for a ASCII test Display Etc. I would recommend to you to change this Pull request to be just for the 16Bit io.

I did create on for the 16 bit I/O but i kind of want to work on these together as some i want to make them work together for example the ports i know someone already made a version but i think i want to implement it with http servers and WebSocket server and have it also be able to send 16bit out. For the moding api i might merge it with this in the end idk

squigglesdev commented:
Hey, I have already made a port implementation (#406 - not using http or websockets, because they're slow) built and [published](https://squigglesdev.itch.io/digital-logic-sim-ports) (updating to 2.1.6 as we speak) that was decided not to be part of the main project, so I don't see the need to make a completely new implementation and try to contribute to the project with them.

And also, as stated earlier, a Modding API is already in the works (#459)

I/O above 8-bit does sound like an interesting idea though!

parshwa282011 commented:
> Hey, I have already made a port implementation (#406 - not using http or websockets, because they're slow) built and [published](https://squigglesdev.itch.io/digital-logic-sim-ports) (updating to 2.1.6 as we speak) that was decided not to be part of the main project, so I don't see the need to make a completely new implementation and try to contribute to the project with them.
> 
> And also, as stated earlier, a Modding API is already in the works (#459)
> 
> I/O above 8-bit does sound like an interesting idea though!

i was thinking http and WebSockets as they are easier to program a extension in python or something like that, and the moding api i am still thinking if i want to do it differently or merge that one.

squigglesdev commented:
You understand that a pull request is not to show what you’re up to right? It’s for changes that everyone will receive. 

parshwa282011 commented:
I added the ports cause I wanted to implement them with 16 bit but mainly I'm probably going to work on the key and ascii displays

pk3zip commented:
Later I’ll make a quick pr w/ 16&32 bit IO

Commit:
Works with Ascii 8bit
The only issue is special charachters are just blank plus 128-255 is also blank

Commit:
Works with Ascii 8bit
The only issue is special charachters are just blank plus 128-255 is also blank

parshwa282011 committed:
Made Ascii 8Bit work

Only issue is special charachters are blank plus anything after 127 in decimal is also blank

BraveCaperCat2 commented:
> I believe that you that you should do one thing at a time as a modding API in already on the way (#459), and to make the reviewing process easier it common for pull requests to only add 1 large feature at a time. For example you would make 1 pull request for 16Bit IO/Ports, 1 for a ASCII test Display Etc. I would recommend to you to change this Pull request to be just for the 16Bit io.

There's already a PR for 16-bit IO. (and 32-bit IO at the same time)

> Later I’ll make a quick pr w/ 16&32 bit IO

There's already one of those. No need to add another.

Commit:
reworked key chip and text display chip

Commit:
reworked key chip and text display chip

Commit:
port http server

Commit:
port http server

Commit:
tons of updates also realistic internet interface

Commit:
tons of updates also realistic internet interface

parshwa282011 committed:
Merge pull request #7 from parshwa282011/squigglesdev-main

merge

Commit:
finally merged all modding api
